lib/request.ts

```diff
@@ -218,6 +219,28 @@ export const request = (
     }
 
     const requestObject = httpRequestFn(options);
+
+    let dropRequestAfterTimeout: NodeJS.Timer;
+    if (options.dropRequestAfter) {
+      dropRequestAfterTimeout = setTimeout(() => {
+        abortCallback();
+      }, options.dropRequestAfter);
+    }
+
+    function abortCallback() {
+      clearTimeout(dropRequestAfterTimeout);
+      if (!hasRequestEnded) {
+        requestObject.abort();
+        const err = new UserTimeoutError(options, timings);
+        logEvent(EventSource.HTTP_REQUEST, EventName.ERROR, err.message);
```

Comment:
Do we want to mark aborted requests as timeouts when aborting this request because there was a successful request?

Reply:
Logging is one thing, but we also need to make sure that this doesn't open the circuit breaker when aborting because of another successful request.

Reply:
Once one on the request has succeed, retryOperation will be marked as `resolved` which will bypass any reject "bubble", see: 

```javascript
catch((error: ServiceClientError) => {
  if (retryOperation.isResolved()) {
    return;
   }
```

So actually clients are not even aware of the result of this promise. There is a vicious side effect though.: this request is actually never resolved or rejected: is it a problem?

Reply:
Aha, so the circuit breaker command's `success()` or `failure()` are not called. Then, it won't make false `failure()` calls at least!

But isn't the circuit breaker designed with an assumption that one of `success()` or `failure()` is called when an operation ends? It seems that [the circuit breaker thinks that it is a timeout if `success()` or `failure()` are not called](https://github.com/zalando-incubator/perron/blob/96c8d3ae92037a820cca1dd1d617608b4063990c/lib/circuit-breaker.ts#L154-L179). And timeouts contributes to error rate calculation in the circuit breaker, which means it can open the circuit breaker without actual errors.

Reply:
So, we should stop the timeout timer in the circuit breaker when aborting requests because of another successful request. Because calling `success()` will lower the actual error rate, how about creating another function (`cancel` or something) in `Command`?

We may want to mark failures in some cases though. For example, if the first request doesn't respond and the second request succeeds, it makes more sense to mark the first request as a failure.

Reply:
Wow very good catch! Will evaluate adding this new function

Reply:
Ok this approach seems to work. I have to write more tests now to cover all the different cases:
- both requests work
- first fails
- second fails
- first timeouts
- second timeouts
- both fail
- both timeouts